nrf_802154_sl/include/fem/nrf_fem_protocol_api.h

```diff
@@ -0,0 +1,281 @@
+ */
+
+/**
+ * @brief Protocol interface for Power Amplifier (PA) and Low Noise Amplifier (LNA).
+ *
+ * This module enables toggling of GPIO pins before and after the radio transmission and the radio reception
+ * in order to control a Power Amplifier or a Low Noise Amplifier, or both.
+ *
+ * The application must first provide PA and LNA device-specific configuration parameters to this module.
+ * The protocol must then provide PA and LNA protocol configuration parameters before it can use the functionality.
+ *
+ * When the PA/LNA module is configured, the stack can call the provided enable functions before radio activity
+ * to enable the PA or LNA timer configurations for the upcoming radio activity.
+ * By default, PA/LNA is automatically deactivated on the radio DISABLED event.
+ */
+
+#ifndef NRF_FEM_PROTOCOL_API_H__
+#define NRF_FEM_PROTOCOL_API_H__
+
+#include <nrf.h>
+
+#if defined(NRF52_SERIES)
+#include "hal/nrf_ppi.h"
+#elif defined(NRF53_SERIES)
+#include "hal/nrf_dppi.h"
+#else
+#error Unsupported chip
+#endif
+
+typedef enum
+{
+    NRF_802154_FAL_PA  = 1 << 0,
+    NRF_802154_FAL_LNA = 1 << 1,
+    NRF_802154_FAL_ALL = NRF_802154_FAL_PA | NRF_802154_FAL_LNA
+} nrf_fal_functionality_t;
+
+/**
+ * @brief PA and LNA activation event types.
+ */
+typedef enum
+{
+    NRF_802154_FAL_EVENT_TYPE_TIMER,
+    NRF_802154_FAL_EVENT_TYPE_GENERIC,
+    NRF_802154_FAL_EVENT_TYPE_PPI,
+} nrf_802154_fal_event_type_t;
+
+/**
+ * @brief Frontend Abstraction Layer event.
+ *
+ * The event can be a Timer Compare event or an Any event.
+ * Register value is only used for the Timer Compare event and should only contain the timer value relative to the Timer Compare event.
+ */
+typedef struct
+{
+    union
+    {
+        struct
+        {
+            NRF_TIMER_Type * p_timer_instance; /* Pointer to a 1-us resolution timer instance. */
+
+            struct
+            {
+                uint32_t start;           /* Timer value when the FEM can start preparing PA/LNA. */
+                uint32_t end;             /* Timer value at which the PA/LNA have to be prepared. Radio operation shall start at this point. */
+            }       counter_period;
+            uint8_t compare_channel_mask; /* Mask of the compare channels that can be used by the FEM to schedule its own tasks. */
+        } timer;
+        struct
+        {
+            uint32_t register_address;        /* Address of event register. */
+        } generic;
+    }                           event;
+    bool                        override_ppi; /* False to ignore the PPI channel below and use the one set by application. True to use the PPI channel below. */
+    uint8_t                     ppi_ch_id;    /* PPI channel to be used for this event. */
+    nrf_802154_fal_event_type_t type;         /* Type of event source. */
+} nrf_802154_fal_event_t;
+
+/**
+ * @brief Sets up PA using the provided events for the upcoming radio transmission.
```

Comment:
This is too much for doxygen doc in a header file, should be moved to RST documentation in the future. Make sure to follow up with @FrancescoSer